asst2/CODE/open_address_ht.c

```diff
@@ -0,0 +1,51 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "open_address_ht.h"
+
+void createTable(void* ptr) {
+  MEM_LIMIT = 8388608/sysconf(_SC_PAGESIZE);
+  VIRT_LIMIT = 25165824/sysconf(_SC_PAGESIZE);
+  HT_SIZE = MEM_LIMIT;
+  ht_entry* table = ptr;
+  for (int i = 0; i < HT_SIZE; ++i) {
+    table[i] = (ht_entry) {HT_NULL_KEY_MASK | HT_NULL_VAL_MASK};
+  }
+}
+
+ht_val ht_put(OpenAddrHashtable table, ht_entry entry) {
+  int hash = (key(entry).mdata % HT_SIZE);
+  while (!(table[hash].mdata & HT_NULL_VAL_MASK) && key(table[hash]).mdata != hash)
+    hash = (hash + 1) % HT_SIZE;
+  ht_val oldval = val(table[hash]);
+  table[hash] = entry;
+  return oldval;
+}
+
+ht_val ht_get(OpenAddrHashtable table, ht_key getkey) {
```

Comment:
again have pid and where process thinks as params rather than having to precast into hashtable types, black box it... ? in the other functions too, i feel like black boxing the actual ht values and accepting malloc / scheduler values would be better?

Reply:
ok I guess I'll just turn it into a struct with the same fields